File: leetcode/27. Remove Element/solution1.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        if (nums.size() == 0) {
            return 0;
        }
        
        int i = 0;
        
        while (i < nums.size() && nums[i] != val) {
            i += 1;
        }
        
        // nums[i] = val now
        
        int j = i + 1;
        int cnt = i;
        
        for (; j < nums.size(); ++j) {
            if (nums[j] != val) {
                nums[i] = nums[j];
                cnt += 1;
                i += 1;
            }
        }
        
        return cnt;
    }
};

File: leetcode/18. 4Sum/solution1.py
class Solution:
    def fourSum(self, nums: List[int], target: int) -> List[List[int]]:
        
        result_set = []
        
        if len(nums) < 4:
            return []
        
        nums.sort()
        
        prev_element_i = None
        
        for i in range(len(nums) - 3):
            
            element_i = nums[i]
            
            if element_i == prev_element_i:
                continue
            
            prev_element_i = element_i
            
            prev_element_j = None
            
            for j in range(i + 1, len(nums) - 2):
                
                element_j = nums[j]
                
                if prev_element_j == element_j:
                    continue
                
                prev_element_j = element_j
                
                k = j + 1
                l = len(nums) - 1
                
                prev_element_k = None
                prev_element_l = None
                
                while k < l:
                    element_k = nums[k]
                    element_l = nums[l]
                    
                    curr_sum = element_i + element_j +\
                               element_k + element_l
                    
                    if curr_sum == target:
                        
                        if prev_element_k == element_k and prev_element_l == element_l:
                            k += 1
                            l -= 1
                            continue
                        
                        result_set.append([element_i, element_j,
                                           element_k, element_l])
                        
                        prev_element_k = element_k
                        prev_element_l = element_l
                        
                        k += 1
                        l -= 1
                        
                    elif curr_sum > target:
                        l -= 1
                    
                    else:
                        k += 1
        
        return result_set
